BotBrownCore/Workers/ConfigurationWatcher.cs

```diff
@@ -0,0 +1,62 @@
+﻿namespace BotBrown.Workers
+{
+    using BotBrown.Configuration;
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    class ConfigurationWatcher
+    {
+        private const int ConfigurationWatchWaitTime = 5000;
+        private IEventBus bus;
+        private IConfigurationManager configurationManager;
+        private ILogger logger;
+        private Dictionary<string, DateTime> configFiles = new Dictionary<string, DateTime>();
+
+        public ConfigurationWatcher(IEventBus bus, IConfigurationManager configurationManager, ILogger logger)
+        {
+            this.bus = bus;
+            this.configurationManager = configurationManager;
+            this.logger = logger;
+        }
+
+        public Task<bool> StartWatch(CancellationToken cancellationToken)
+        {
+            foreach (var file in Directory.EnumerateFiles(Directory.GetCurrentDirectory()))
+            {
+                if (!file.EndsWith(".json"))
+                {
+                    continue;
+                }
+
+                configFiles.Add(file, File.GetLastWriteTime(file));
+            }
+
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                var now = DateTime.Now;
+                foreach (var file in Directory.EnumerateFiles(Directory.GetCurrentDirectory()))
+                {
+                    if (!file.EndsWith(".json"))
+                    {
+                        continue;
+                    }
+
+                    var lastWriteTime = File.GetLastWriteTime(file);
+                    if (configFiles.ContainsKey(file) && configFiles[file] < lastWriteTime)

```

Comment:
We could maybe use a FileSystemWatcher here. I haven't used it before so I am not sure but it might help us here? I'll still accept this pull request and add it later if you think it would help.

https://docs.microsoft.com/de-de/dotnet/api/system.io.filesystemwatcher?view=netframework-4.7.2